UseCases.md

```diff
@@ -0,0 +1,100 @@
+This file describes exact behavior of methods for different edge cases and 
+explains general logic. This description covers the behavior of get_tld, 
+get_tld_unsafe, get_sld, get_sld_unsafe, split_domain, split_domain_unsafe
+
```

Comment:
@vadym-t suggest to add here a why sentence: 
Unsafe versions of the methods will significantly save resources on large-scale applications of the library where the data has already been converted to lowercase and missing data has a None value. This can be done in Spark/Dask, for example, and result in a significant reduction in computational resources. For adhoc usage, the original functions are sufficient.

Reply:
Added